react/utils/normalize.ts

```diff
@@ -67,14 +77,149 @@ export function changeImageUrlSize(
-      commertialOffer.AvailableQuantity != null &&
-      commertialOffer.AvailableQuantity > 0
+function getPriceFromSeller(seller: ProductTypes.Seller) {
+  return seller.commertialOffer.Price
+}
+
+function isAvailable({
+  commertialOffer: { AvailableQuantity },
+}: ProductTypes.Seller) {
+  return AvailableQuantity != null && AvailableQuantity > 0
+}
+
+/**
+ * @description
+ * Find the lowest/highest `Price` between sellers
+ * @param item an SKU
+ * @param condition the condition enum used for comparison functions
+ * */
+function getBestPrice(item: ProductTypes.Item, condition: PriceConditionRule) {
+  // If there's only 1 seller, avoid filtering
+  const { sellers } = item
+
+  if (sellers.length === 1) return sellers[0].commertialOffer.Price
+
+  const availableSellers = sellers.filter(isAvailable)
+  const allPrices = availableSellers.map(getPriceFromSeller)
+
+  let itemPrice
+
+  if (condition === 'highest') {
+    itemPrice = allPrices.reduce((max, price) => (price > max ? price : max))
+  } else {
+    itemPrice = allPrices.reduce((min, price) => (price < min ? price : min))
+  }
+
+  return itemPrice
+}
+
+/**
+ * @description
+ * Retrieves the SKU with the best price considering a provided condition
+ * @param items an array of SKUs
+ * @param condition the condition [highest|lowest]
+ * */
+function findSKUByPrice(
+  items: ProductTypes.Item[],
+  condition: PriceConditionRule
+) {
+  // If none or only 1 sku is available, avoid reducing
+  const availableItems = items.filter(findAnyAvailable)
+
+  if (availableItems.length === 0) return items[0]
+  if (availableItems.length === 1) return availableItems[0]
+
+  const bestPrices = availableItems.map((item) => getBestPrice(item, condition))
+  let itemToReturn
+
+  if (condition === 'highest') {
+    itemToReturn = bestPrices.reduce(
+      (maxIndex, currentPrice, currentIndex) =>
+        currentPrice > bestPrices[maxIndex] ? currentIndex : maxIndex,
+      0
+    )
+  } else {
+    itemToReturn = bestPrices.reduce(
+      (minIndex, currentPrice, currentIndex) =>
+        currentPrice < bestPrices[minIndex] ? currentIndex : minIndex,
+      0
+    )
+  }
+
+  return availableItems[itemToReturn]
+}
+
+/**
+ * @summary
+ * Used to return the first available SKU
+ * */
+function getAvailableProduct({ sellers }: ProductTypes.Item) {
+  return sellers.find(isAvailable)
+}
+
+/**
+ * @summary
+ * Used to probe if at least 1 seller has stock
+ * */
+function findAnyAvailable({ sellers }: ProductTypes.Item) {
+  return sellers.some(isAvailable)
+}
+
+/**
+ * @description
+ * Attempts to retrieve the specific SKU using a custom Product (field)
+ * in the Catalog. The specification has the value of the SKU ID
+ * that will be looking for. If not found by any reason, we default
+ * to finding the preferred SKU
+ * @param items an array of SKUs
+ * @param defaultSKUspec the ID used to find the SKU
+ * @param preferenceFallback the logic for finding an SKU if the ID fails
```

Comment:
Ain't an English expert, but Grammarly warned about it.
```suggestion
 * Attempts to retrieve the specific SKU using a custom Product (field)
 * in the Catalog. The specification has the value of the SKU ID
 * that will be looking for. If not found for any reason, we default
 * to finding the preferred SKU
 * @param items an array of SKUs
 * @param defaultSKUspec the ID used to find the SKU
 * @param preferenceFallback the logic for finding an SKU if the ID fails
```